Deprecate ez_http_tag_location

adriendupuis opened the pull request:
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | N/A
| **Type**           | Improvement/Misc
| **Target version** | `master`
| **BC breaks**      | no
| **Doc needed**     | https://github.com/ezsystems/developer-documentation/pull/1297

Mark ez_http_tag_location as deprecated and ez_http_cache_tag_location as its alternative. If 2.5 BC is “to be cleaned up”, one should be informed to not using this and replace it.

**TODO**:
- [x] Implement feature / fix a bug.
- [ ] Implement tests + specs and passing (`$ composer test`)
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

adriendupuis committed:
Deprecate ez_http_tag_location

Mark ez_http_cache_tag_location as deprecated and ez_http_cache_tag_location as its alternative.

adamwojs commented on src/Twig/ContentTaggingExtension.php:
```diff
@@ -43,7 +43,8 @@ public function getFunctions()
                 'ez_http_tag_location',
-                [$this, 'tagHttpCacheForLocation']
+                [$this, 'tagHttpCacheForLocation'],
+                ['deprecated' => true, 'alternative' => 'ez_http_cache_tag_location']
```
```suggestion
                [
                    'deprecated' => true, 
                    'alternative' => 'ez_http_cache_tag_location',
                ]
```

adriendupuis committed:
Format src/Twig/ContentTaggingExtension.php

Co-authored-by: Adam Wójs <adam@wojs.pl>

vidarl commented on src/Twig/ContentTaggingExtension.php:
```diff
@@ -43,7 +43,11 @@ public function getFunctions()
+                ]
             ),
             new TwigFunction(
                 'ez_http_tag_relation_ids',
```
Same goes for the `ez_http_tag_relation_location_ids` function that should be deprecated and replaced with `ez_http_tag_relation_location_ids`.  Doesn't make much sense that only one of these functions are prefixed `_ez_http_cache_`, while the others are prefixed `ez_http_`

adriendupuis replied:
@vidarl Good remark about naming homogeneity but creating new Twig functions wasn't the initial goal of this PR. Could it be kept for another PR?

vidarl replied:
To me it makes no sense to deprecate `ez_http_tag_location`, unless we also deprecate the other `ez_http_tag*` functions.
If that is done in this PR or in a separate one is of no importance to me

adriendupuis replied:
You are right. Saying that `ez_http_` prefix is deprecated in favor of `ez_http_cache_` but having this true for only one, it's not serious. I'll add the other two.

adriendupuis replied:
@vidarl Done; two new Twig functions and deprecation of the old two.

adriendupuis committed:
ez_http_cache_tag_ prefix on every Twig function

Add ez_http_cache_tag_relation_ids and ez_http_cache_tag_relation_location_ids along ez_http_cache_tag_location for consistency.

Deprecate all

vidarl commented:
@lserwatka So, this will be deprecated in 4.0 then (and possibly removed in 5.0) ?

lserwatka commented:
This is already deprecated, we need to add a note in 3.3 and remove it in 4.0 as discussed offline.

adamwojs commented:
@adriendupuis CI is reporting that code style needs to be fixed here. Could you please run `composer fix-cs` command, and then commit and push it's result? Thank you in advance,

dabrt commented:
Dear gentlemen, when exactly were the two functions deprecated?
I mean `ez_http_tag_relation_ids()` and `ez_http_tag_relation_location_ids()`
Was it starting from 3.0 or 3.3?

vidarl commented:
Communication wise, these where deprecated in 3.3.0
Technically, they will be deprecated in 3.3.2 ( if this PR gets merged by then )

adamwojs committed:
Fixed code style issues

vidarl commented:
@adamwojs : ready for merge, right?
